shared/src/test/diff/ucs/HygienicBindings.mls

```diff
@@ -0,0 +1,164 @@
+:NewDefs
+
+type Option[out T] = None | Some[T]
+module None
+class Some[out T](value: T)
+//│ type Option[T] = Some[T] | None
+//│ module None()
+//│ class Some[T](value: T)
+
+type Either[A, B] = Left[A] | Right[B]
+class Left[A](leftValue: A)
+class Right[B](rightValue: B)
+//│ type Either[A, B] = Left[A] | Right[B]
+//│ class Left[A](leftValue: A)
+//│ class Right[B](rightValue: B)
+
+type List[out A] = Nil | Cons[A]
+module Nil
+class Cons[out A](head: A, tail: List[A])
+//│ type List[A] = Cons[A] | Nil
+//│ module Nil()
+//│ class Cons[A](head: A, tail: List[A])
+
+fun h0(a) =
+  if
+    a is Some(Left(y)) then y
+    a is Some(Right(z)) then z
+    a is None then 0
+//│ fun h0: forall 'leftValue. (None | Some[Left['leftValue] | Right['leftValue]]) -> (0 | 'leftValue)
+
+// FIXME: Precise scrutinee identification (easy)
+// This seems fine. But the subtrees are not merged.
+fun h1(a) =
+  if
+    a is Some(x) and x is Left(y) then y
+    a is Some(y) and y is Right(z) then z
+    a is None then 0
+//│ fun h1: forall 'leftValue. (None | Some[Right['leftValue]]) -> (0 | 'leftValue)
+
+// FIXME: This is the desugared version of the test case above.
```

Comment:
Does this one need to be fixed?! I don't think one should expect the UCS to merge explicit let bindings.

Reply:
We don't need to merge them, but I think the exhaustiveness checker should be aware of explicit let bindings which refer to the same class parameter.

Reply:
I've added this to backlog issue.